lib/prefab/config_client.rb

```diff
@@ -171,6 +176,36 @@ def load_configs(configs, source)
       finish_init!(source, project_id)
     end
 
+    def cache_path
+      return @cache_path unless @cache_path.nil?
+
+      @cache_path ||= File.join(ENV.fetch('XDG_CACHE_HOME', File.join(Dir.home, '.cache')), "prefab.cache.#{@base_client.options.api_key_id}.json")
+      FileUtils.mkdir_p(File.dirname(@cache_path))
+      @cache_path
+    end
+
+    def cache_configs(configs)
+      return unless @options.use_local_cache && !@options.is_fork
+      File.open(cache_path, "w") do |f|
+        f.flock(File::LOCK_EX)
+        f.write(PrefabProto::Configs.encode_json(configs))
```

Comment:
The lock prevents multiple writes, but this still needs the atomic-write pattern so non one reads a half written file